### DISCUSSION TITLE
Import Issue

### DISCUSSION THREAD

**Discussion by pnbd** [photoprism/photoprism#2840] [Q&A]
Hi guys, first of all, thank you so much for the amazing work. 
I just started to get a problem since I made a pull for the last version and something strange started happening. So, when I put files in the import folder and press import, instead of moving the files, like normally, to the year and month folder, it puts them inside a year/month/day folder, inside the import folder itself... Then, if I press import again, it deletes everything because it says that the files were already imported.

**Comment by lastzero** [comment]
What last version? If you are not sure, you should use GitHub Discussions first.

**Comment by lastzero** [comment]
Tested it, the logs look as expected (maybe you use another tool that creates these subfolders?):

```
import: moving main jpg file 2011/10/20111002_120138_783F539B.jpg to 2011/10/20111002_120138_783F539B.jpg 
```

**Reply by pnbd** [reply]
Sorry if this is not the version you were asking for, but I believe, I'm on build 220901-f493607b0
Other than that, I manually moved the videos and photos manually to the import folder and clicked import, with the move checkbox turned on.

**Reply by lastzero** [reply]
If there was something wrong with this old version, probably someone else would have reported it already?

That files are deleted from the import folder without moving them can happen if they already exist in the index or physically on disk:

> When [importing](https://docs.photoprism.app/user-guide/library/import/), files are first copied or moved from a temporary folder to the originals folder. In the process, duplicates are automatically skipped. "Move" also deletes them in the source directory as if they were successfully moved.

See the related documentation:
- https://docs.photoprism.app/user-guide/library/duplicates/
- https://docs.photoprism.app/user-guide/library/import/

**Reply by lastzero** [reply]
PS: You should make sure that the *import* folder is NOT INSIDE the *originals* folder and vice versa.

**Reply by pnbd** [reply]
It's strange, I tried it on another file and I couldn't reproduce the behavior... I turned on debug log, I anexed the log below. Hopefully it'll help. And sorry if the mistake is on my part.

[photoprism.txt](https://github.com/photoprism/photoprism/files/9912822/photoprism.txt)

**Reply by pnbd** [reply]
Hi there, sorry to come back to this issue, but it happened to me again. From what I can gather I imported photos through webdav into the import folder. When I went to check it, photoprism had already indexed those photos. The problem is when I press import with the Move tag  selected. It starts deleting all the photos in the import folder because they are already, in the import folder... It doesn't always happen, but I'm sure it as to do with the indexing of the photos in the import folder. Is there a way to stop the indexing of photos in the index folder?

**Reply by lastzero** [reply]
Did you read this? https://docs.photoprism.app/known-issues/#nested-storage-folder

I don't know how else something like that could happen unless you run custom background jobs.

**Reply by pnbd** [reply]
Well I have docker-compose.yml inside /opt/photoprism and the storage and database folder on that same folder. Although, I do have the import folder inside the originals folder which is in another completely diferent folder and disk /media/cofre/etc ... Maybe that is the problem. Do you think that adding a .ppignore file with the import folder might fix the issue? Or is it better to take the import folder from inside the originals?

**Reply by lastzero** [reply]
Use different folders and don't nest them in any way. A ppignore file may help, but when things get overly complicated, it's really difficult to provide support. Also note that an ignore file does not affect already indexed / imported files.

**Reply by pnbd** [reply]
Yeah... Now I understand the problem... Thank you so much. Is there a way for me to find out how many files might be indexed in the import folder, in order to find out how many files are in this situation?

**Reply by lastzero** [reply]
If you need specific counts, the only way that I can think of would be checking the files table in the database. Otherwise you can use the file browser in Library > Originals to see what is currently indexed.

**Reply by pnbd** [reply]
I'll try the database option, because in Library > Originals nothing appears... Thank you very much for all the help and sorry for the stupidity. As a friend usually says, the problem with software, 99% of the times is in between the computer and the chair.

**Reply by lastzero** [reply]
Updated the docs so it is clear you must not place the import folder in originals: https://docs.photoprism.app/getting-started/docker-compose/#photoprismimport
